CounterApp/App/AppView.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  AppView.swift
+//  CounterApp
+//
+//  Created by Wenjuan Li on 2022/7/7.
+//
+
+import ComposableArchitecture
+import StoreKit
+import SwiftUI
+
+struct AppView: View {
+    let store: Store<AppState, AppAction>
+
+    var body: some View {
+        WithViewStore(store.scope(state: \.viewState)) { viewStore in
+            NavigationView {
+                VStack {
+                    Text("\(viewStore.counter)")
+
+                    NavigationLink(
+                        "Goto edit page",
+                        destination: CounterView(
+                            store: self.store.scope(
+                                state: { appState in appState.counter },
+                                action: { localAction in AppAction.counter(localAction) }
+                            )
+                        )
+                    )
+                    .padding()
+
+                    Button("Show Lock View") {
+                        viewStore.send(.lock(LockAction.setSheet(isPresented: true)))
+                    }
+                }
+                .font(Font.title2)
+            }
+            .sheet(
+                isPresented: viewStore.binding(
+                    get: \.isShowLockView,
+                    send: { localState in
+                        AppAction.lock(LockAction.setSheet(isPresented: localState))
+                    }
+                )
+            ) {
+                LockView(
+                    store: self.store.scope(
+                        state: { appState in appState.lock },
+                        action: { localAction in AppAction.lock(localAction) }
+                    )
+                )
+            }
+        }
+    }
+}
+
+extension AppView {
+    struct ViewState: Equatable {
+        var counter: String
+        var isShowLockView: Bool
+    }
```

Comment:
This is great! One thing here is that we usually want to make explicit that these won't change, so we tend to declare them as `let`.